### PR TITLE
Add support for new versions of TFEL and MGIS

### PR DESCRIPTION
**Pull request by thelfer**

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the new version branches and sha256, rest LGTM.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
The changes look fine but I just noticed there are unrelated commits.  This typically happens when a rebase has issues.

Please reduce the number of commits to only those that apply to this PR.

**Comment by tldahlgren** [comment]
> The changes look fine but I just noticed there are unrelated commits. This typically happens when a rebase has issues.
> 
> Please reduce the number of commits to only those that apply to this PR.

After looking at some of the commits I might be mistaken but there are changes that no longer appear to be in the final files. 

Perhaps the other contributors (@bernhardkaindl @sethrj) weigh in on this?

Ping me if this PR (and all associated commits) are intentional.

**Comment by sethrj** [comment]
@tldahlgren I think the problem is that @thelfer is submitting directly from the 'develop' branch on his fork, which I guess dependabot is trying to update? I'm not a contributor to this so I'm not sure why I'm tagged. Since the commits will be squashed I think it's fine to have the bajillion "merge develop" there.

File: var/spack/repos/builtin/packages/tfel/package.py [hunk 1]
```diff
@@ -88,6 +105,12 @@ class Tfel(CMakePackage):
     version("3.0.0", sha256="b2cfaa3d7900b4f32f327565448bf9cb8e4242763f651bff8f231f378a278f9e")
     version("2.0.4", sha256="cac078435aad73d9a795516f161b320d204d2099d6a286e786359f484355a43a")
 
+    variant(
+        "build_type",
+        default="Release",
+        description="The build type to build",
+        values=("Debug", "Release"),
+    )
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/tfel/package.py]: This shouldn't be necessary because the CMakePackage includes the variant already.

Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: @sethrj Thanks for the tip. Fixed.

**Comment by thelfer** [comment]
> @tldahlgren I think the problem is that @thelfer is submitting directly from the 'develop' branch on his fork, which I guess dependabot is trying to update? I'm not a contributor to this so I'm not sure why I'm tagged. Since the commits will be squashed I think it's fine to have the bajillion "merge develop" there.

@tldahlgren, @sethrj I basically made in the develop branch of my fork:

```bash
git fetch upstream
git merge upstream/develop
# edit the two files
git commit -a -m "..."
git push
```

To be honest, I have no clue how to suppress the additional commits (nor why they appear, some were done during the last update of the packages, the other are automatic form dependabot)

**Comment by tldahlgren** [comment]
> @tldahlgren .. so I'm not sure why I'm tagged. Since the commits will be squashed I think it's fine to have the bajillion "merge develop" there.

I tagged you because your account showed up in the commits.

**Comment by sethrj** [comment]
OK, those were the commits that accidentally showed up but are gone now. Looks like everything is good.

**Comment by tldahlgren** [comment]
> > @tldahlgren I think the problem is that @thelfer is submitting directly from the 'develop' branch on his fork, which I guess dependabot is trying to update? I'm not a contributor to this so I'm not sure why I'm tagged. Since the commits will be squashed I think it's fine to have the bajillion "merge develop" there.
> 
> @tldahlgren, @sethrj I basically made in the develop branch of my fork:
> 
> ```shell
> git fetch upstream
> git merge upstream/develop
> # edit the two files
> git commit -a -m "..."
> git push
> ```
> 
> To be honest, I have no clue how to suppress the additional commits (nor why they appear, some were done during the last update of the packages, the other are automatic form dependabot)

We have some documentation on branching and rebasing at https://spack.readthedocs.io/en/latest/contribution_guide.html#git-workflows that might be helpful.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**